core/src/test/java/org/apache/servicecomb/core/provider/consumer/TestConsumerProviderManager.java

```diff
@@ -96,4 +108,42 @@ public void createReferenceConfig_config() {
     Assert.assertEquals("1.0.0+", referenceConfig.getMicroserviceVersionRule().getVersionRule().getVersionRule());
     Assert.assertEquals(Const.RESTFUL, referenceConfig.getTransport());
   }
+
+  @Test
+  public void createReferenceConfig_ProviderNotFound() {
+    EventBus eventBus = new EventBus();
+    AppManager appManager = new AppManager(eventBus);
+
+    ConsumerProviderManager consumerProviderManager = new ConsumerProviderManager();
+    consumerProviderManager.setAppManager(appManager);
+
+    new Expectations(RegistryUtils.class) {
+      {
+        RegistryUtils.findServiceInstances(anyString, anyString, DefinitionConst.VERSION_RULE_ALL, null);
+        result = Collections.emptyList();
+      }
+    };
+
+    new MockUp<MicroserviceVersionRule>() {
+      @Mock
+      String getAppId() {
+        return "aId";
+      }
+
+      @Mock
+      String getMicroserviceName() {
+        return "ms";
+      }
+    };
+
+    try {
+      ReferenceConfig referenceConfig = consumerProviderManager.createReferenceConfig("app:ms");
```

Comment:
unused variable. and if this method not throw exception, test should be failed.

Reply:
fixed, please review again.